src/main/java/org/folio/innreach/domain/service/impl/InnReachTransactionActionServiceImpl.java

```diff
@@ -304,6 +304,23 @@ public void transferItemHold(UUID transactionId, String itemBarcode) {
     eventPublisher.publishEvent(MoveRequestEvent.of(transaction, item));
   }
 
+  @Override
+  public void finalCheckInItemHold(UUID transactionId, UUID servicePointId) {
+    var transaction = fetchTransactionById(transactionId);
```

Comment:
it would be more safe to restrict transaction type
```suggestion
    var transaction = fetchTransactionOfType(transactionId, ITEM);
```

Reply:
thanks for useful info